File: controller/Controller/src/hkccpacmanrobot/controller/Controller.java
package hkccpacmanrobot.controller;

import hkccpacmanrobot.utils.message.Messenger;
import hkccpacmanrobot.utils.message.MovementCommand;

import java.io.IOException;
import java.net.Socket;

public class Controller {
    public final Messenger<MovementCommand> messenger;

    public Controller(Socket socket) {
        messenger = new Messenger<MovementCommand>(socket);
    }

    public Controller(String hostname, int port) {
        try {
            Socket socket = new Socket(hostname, port);
            messenger = new Messenger<MovementCommand>(socket);
        } catch (IOException e) {
            throw new RuntimeException(e.toString());
        }
    }

    public void sendCommand(MovementCommand command) {
        messenger.sendMessage(command);
    }
}

File: controller/Controller/src/hkccpacmanrobot/controller/Controller.scala
package hkccpacmanrobot.controller

import hkccpacmanrobot.utils.message.{Device, Messenger, MovementCommand}

/**
 * Created by beenotung on 3/23/15.
 */
class Controller extends Device {
  val movementCommandMessenger: Messenger[MovementCommand] = Messenger.create[MovementCommand](MovementCommand)
  var shouldSend = true

  def sendCommand(command: MovementCommand) {
    movementCommandMessenger.sendMessage(command)
  }

  override def gameSetup: Unit = {
    shouldSend = false
  }

  override def gameResume: Unit = {
    shouldSend = true
  }

  override def gamePause: Unit = {
    shouldSend = false
  }

  override def gameStop: Unit = {
    shouldSend = false
  }

  override def gameStart: Unit = {
    shouldSend = true
  }
  override def run={
    super.run
    new Thread(new Runnable {
      override def run(): Unit = {

      }
    })
  }.start()
}
